### PR TITLE
KAFKA-2462: allow modifying soft limit for open files in Kafka startup script

### PR DESCRIPTION
**Pull request by gwenshap**

### CODE REVIEW

File: bin/kafka-server-start.sh [hunk 1]
```diff
@@ -37,6 +37,10 @@ case $COMMAND in
     EXTRA_ARGS="-daemon "$EXTRA_ARGS
     shift
     ;;
+  --soft-file-limit)
```
Comment (ewencp) [inline bin/kafka-server-start.sh:40]: I think this doesn't work if you provide them in the reverse order (i.e., `--soft-file-limit <number> -daemon`, which I would assume would be possible)? I think you'd need a while loop around this.

File: bin/kafka-server-start.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 if [ $# -lt 1 ];
 then
-	echo "USAGE: $0 [-daemon] server.properties [--override property=value]*"
+	echo "USAGE: $0 [-daemon] [--soft-file-limit number] server.properties [--override property=value]*"
```
Comment (SinghAsDev) [inline bin/kafka-server-start.sh:19]: Just curious, if it would make sense to have it as part of kafka-env.sh, as suggested in KAFKA-1566?

**Comment by guozhangwang** [comment]
@gwenshap Would you like to continue working on this PR?

**Comment by guozhangwang** [comment]
Closing for cleanup.

**Pull request closed without merging**